lib/http.c

```diff
@@ -3624,7 +3666,16 @@ static CURLcode http_on_response(struct Curl_easy *data,
     k->download_done = TRUE;
 
   /* final response without error, prepare to receive the body */
-  return Curl_http_firstwrite(data);
+  result = Curl_http_firstwrite(data);
+
+out:
+  if(last_hd) {
```

Comment:
Surely this should not be done if it already has an error in `result` here?

Reply:
I am aiming or backward compatibility. We did write the end first and then fail. A callback now would no longer see the last header line. Maybe it does not make a difference, but...

Reply:
But that first error could come as an error from the header callback can't it? Meaning that it would then be called again?

Reply:
I think not. Line 3419 sets `last_hd = NULL;` before checking the result.

Reply:
Ah yes, I missed that. Okay, but can you then add a comment explaining why this is called even if it may already have received an error so that future me won't have to ask again? :smiley: